odoo/tests/common.py

```diff
@@ -2427,28 +2427,60 @@ def tagged(*tags):
     def tags_decorator(obj):
         include = {t for t in tags if not t.startswith('-')}
         exclude = {t[1:] for t in tags if t.startswith('-')}
-        obj.test_tags = (getattr(obj, 'test_tags', set()) | include) - exclude # todo remove getattr in master since we want to limmit tagged to BaseCase and always have +standard tag
+        obj.test_tags = (getattr(obj, 'test_tags', OrderedSet()) | include) - exclude # todo remove getattr in master since we want to limmit tagged to BaseCase and always have +standard tag
         return obj
     return tags_decorator
 
 
+def split_parts(command):
+    # split on ',' but not inside '[]'
+    # a regex like r'([^,\[\]]+(?:\[.*?\])*)+' may work but be less easy to maintain and doesn't support nested []
+    parts = []
+    part = ''
+    level = 0
+    for char in command:
+        if level == 0 and char == ',':
+            parts.append(part)
+            part = ''
+        else:
+            part += char
+            if char == '[':
+                level += 1
+            elif char == ']':
+                level -= 1
+        assert level >= 0
+    parts.append(part)
+    assert level == 0
+    return parts
+        
 class TagsSelector(object):
     """ Test selector based on tags. """
-    filter_spec_re = re.compile(r'^([+-]?)(\*|\w*)(?:/(\w*))?(?::(\w*))?(?:\.(\w*))?$')  # [-][tag][/module][:class][.method]
+    tags_parts = (
+        r'([+-]?)',        #operator_re
+        r'(\*|\w*)',       #tag_re
+        r'(?:/(\w*))?',    #module_re
+        r'(?::(\w*))?',    #test_class_re
+        r'(?:\.(\w*))?',   #test_method_re
+        r'(?:\[(.*)\])?',  #parameters
+    )
+    tag_re = ''.join(tags_parts) #[-][tag][/module][:class][.method]
+    filter_spec_re = re.compile(rf'^{tag_re}$')
 
     def __init__(self, spec):
         """ Parse the spec to determine tags to include and exclude. """
-        filter_specs = {t.strip() for t in spec.split(',') if t.strip()}
-        self.exclude = set()
-        self.include = set()
+        #parts = re.findall(r'([^,\[\]]+(?:\[.*?\])*)+', spec) 
+        parts = split_parts(spec)
+        filter_specs = [t.strip() for t in parts if t.strip()]
```

Comment:
The stripping should be done by `split_parts`, no reason to require post-processing of something we hand-processed already.